lcompilers/lpython
# add flag `–show_documentSymbols` for lsp symbol lookup

This PR adds a flag --show-errors which prints the diagnostics from the file to the console. This is used by VSCode Language Server extension to catch the stdout and parse into JSON,

 Steps to test it - 

1. Clone https://github.com/ankitaS11/lpython and check out the` lsp/add_flag/show-error` branch.
2. Compile lpython with LSP flag on:

``` 
conda activate lp # or use your environment name here
cmake -DCMAKE_BUILD_TYPE=Debug -DWITH_LLVM=yes -DWITH_STACKTRACE=yes -DWITH_LFORTRAN_BINARY_MODFILES=no -DWITH_LSP=yes .
cmake --build . -j16
```
3. To get the errors/ warnings (if any) for the given file path in JSON format: run `./src/bin/lpython --show-documentSymbols <file_name>` in terminal. This will be used by LPython’s VSCode extension for document symbol lookup feature.

To Test it in VSCode extension - 

Clone LPython’s VSCode Extension: https://github.com/ankitaS11/lpyth and follow the instructions mentioned in its README.

NOTE: This can only be tested once #863 gets merged.
